subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/testng/TestNGOptions.java

```diff
@@ -88,10 +88,10 @@ public TestNGOptions(ProjectLayout projectLayout) {
     }
 
     /**
-     * Copies the TestNG options.
+     * Copies the options from the source options into the current one.
      * @since 8.0
      */
-    public TestNGOptions(TestNGOptions other) {
+    public void copyFrom(TestNGOptions other) {
```

Comment:
The `addAll` calls are not really correct as we're not doing this in the constructor anymore. We should clear the collections first before populating them again.